### PR TITLE
[FRONT-1499] Config from Chain 

### PR DESCRIPTION
**Pull request by pmichalowski**

### CODE REVIEW

File: src/getters/getConfigFromChain.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * When we will be needing more of the fields, simply expand the return type and add them
+ */
+export const getConfigFromChain = async (): Promise<{
+    maxPenaltyPeriodSeconds: BNish
+    minimumStakeWei: BNish
+}> => {
+    const signer = await getSigner()
+    const token = new Contract(
```
Comment (tumppi) [inline src/getters/getConfigFromChain.ts:15]:
```suggestion
    const config = new Contract(
```

File: src/getters/getConfigFromChain.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        defaultChainConfig.contracts.StreamrConfig,
+        streamrConfigABI,
+        signer,
+    ) as StreamrConfig
+    return {
+        maxPenaltyPeriodSeconds: await token.maxPenaltyPeriodSeconds(),
```
Comment (tumppi) [inline src/getters/getConfigFromChain.ts:21]: This could use some error handling. Calling the contract methods can throw.

**Review by tumppi — CHANGES_REQUESTED** [review]

File: src/getters/getConfigFromChain.ts (hunk 1 above)
Comment (pmichalowski) [reply src/getters/getConfigFromChain.ts:15]: Ah yes - the copy-paste side effect :)

File: src/getters/getConfigFromChain.ts (hunk 2 above)
Comment (pmichalowski) [reply src/getters/getConfigFromChain.ts:21]: I think the only thing we can do is throw an error toast. I'll write something like this in the toast message: "Could not load data from the chain, please try again later"

File: src/getters/getConfigFromChain.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { defaultChainConfig } from './getChainConfig'
+
+/**
+ * When we will be needing more of the fields, simply expand the return type and add them
+ */
+export const getConfigFromChain = async (): Promise<{
+    maxPenaltyPeriodSeconds: BNish
+    minimumStakeWei: BNish
+} | null> => {
+    const signer = await getSigner()
```
Comment (mondoreale) [inline src/getters/getConfigFromChain.ts:15]: Here, the other point where this utility can explode but is outside of that try…catch below.

File: src/network/modals/CreateSponsorshipModal.tsx [hunk 4]
```diff
@@ -107,6 +111,23 @@ export default function CreateSponsorshipModal({
...
         !!initialAmount && !!payoutRate && extensionInDays < minStakeDuration
 
+    const [maxPenaltyPeriod, setMaxPenaltyPeriod] = useState<number>()
+
+    const tooLongMinStakeDuration =
+        !!maxPenaltyPeriod && minStakeDuration > maxPenaltyPeriod
+
+    useEffect(() => {
+        getConfigFromChain().then((config) => {
+            if (isMounted() && config) {
```
Comment (mondoreale) [inline src/network/modals/CreateSponsorshipModal.tsx:121]: Let's avoid `useIsMounted` hook, please. Doesn't hurt here but yeah, just saying.

File: src/services/sponsorships.ts [hunk 5]
```diff
@@ -50,6 +51,12 @@ export async function createSponsorship(
 
     const signer = await getSigner()
 
+    const contractConfig = await getConfigFromChain()
```
Comment (mondoreale) [inline src/services/sponsorships.ts:54]: To me, empty `contractConfig` should make it 💥.

File: src/getters/getConfigFromChain.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    ) as StreamrConfig
+    try {
+        return {
+            maxPenaltyPeriodSeconds: await config.maxPenaltyPeriodSeconds(),
+            minimumStakeWei: await config.minimumStakeWei(),
+        }
+    } catch (e) {
+        errorToast({
+            title: 'Could not load data from the chain, please try again later.',
+        })
```
Comment (mondoreale) [inline src/getters/getConfigFromChain.ts:29]: Not sure if I like this toast. Later we throw if `getConfigFromChain` gives a `null`. If we decide to toast about it there then we have 2 toasts, one of which is uncontrollable from the outside.

Maybe drop the `try…catch`?

File: src/network/getters/getTokenAndBalanceForSponsorship.ts [hunk 7]
```diff
@@ -15,7 +15,11 @@ export const getTokenAndBalanceForSponsorship = async (
     const tokenSymbolFromConfig = getCoreConfig().sponsorshipPaymentToken as string
     const tokenInformation = await getTokenInformation(
         defaultChainConfig.contracts[tokenSymbolFromConfig],
+        defaultChainConfig.id,
     )
+
+    // TODO it fails if user in not on the default chain (local or Polygon on prod) - needs refactor
+
```
Comment (mondoreale) [inline src/network/getters/getTokenAndBalanceForSponsorship.ts:22]: Won't `networkPreflight` solve it?

File: src/network/modals/CreateSponsorshipModal.tsx [hunk 8]
```diff
@@ -107,6 +111,23 @@ export default function CreateSponsorshipModal({
...
+                setMaxPenaltyPeriod(
+                    toBN(config.maxPenaltyPeriodSeconds.toString())
```
Comment (mondoreale) [inline src/network/modals/CreateSponsorshipModal.tsx:123]:
```suggestion
                    toBN(config.maxPenaltyPeriodSeconds)
```

File: src/network/modals/CreateSponsorshipModal.tsx [hunk 9]
```diff
@@ -217,13 +238,23 @@ export default function CreateSponsorshipModal({
                 <Section>
                     <WingedLabelWrap>
                         <Label>Minimum time operators must stay staked</Label>
-                        {invalidMinStakeDuration && (
-                            <ErrorLabel>
-                                The value is higher than the duration of the sponsorship
-                            </ErrorLabel>
-                        )}
+                        <div className="error-wrap">
```
Comment (mondoreale) [inline src/network/modals/CreateSponsorshipModal.tsx:241]: Let's avoid inline handmade classes. Use a separate styled component instead.

File: src/services/sponsorships.ts [hunk 10]
```diff
@@ -65,7 +72,7 @@ export async function createSponsorship(
     const data = defaultAbiCoder.encode(
         ['uint', 'uint32', 'uint32', 'string', 'string', 'address[]', 'uint[]'],
         [
-            parseEther('60'), // initialMinimumStakeWei - hardcoded for now
+            contractConfig.minimumStakeWei, // initialMinimumStakeWei
```
Comment (mondoreale) [inline src/services/sponsorships.ts:75]: ❤️

**Review by mondoreale — COMMENTED** [review]
Added my 2¢. Cheers!

File: src/network/getters/getTokenAndBalanceForSponsorship.ts (hunk 7 above)
Comment (pmichalowski) [reply src/network/getters/getTokenAndBalanceForSponsorship.ts:22]: Probably - but the issue is that we are calling it while displaying the list of sponsorships - I made a ticket about it so we can eventually discuss how to approach it on our Monday call

File: src/network/modals/CreateSponsorshipModal.tsx (hunk 4 above)
Comment (pmichalowski) [reply src/network/modals/CreateSponsorshipModal.tsx:121]: Why? It helps to handle those situations where you'd try to update a component that was unmounted.

File: src/getters/getConfigFromChain.ts (hunk 6 above)
Comment (pmichalowski) [reply src/getters/getConfigFromChain.ts:29]: Ok I'll try to improve it somehow. To be honest I'm kinda lacking some well-defined error-handling policy that we could be always referring to. We could speak about it on Monday

Comment (pmichalowski) [reply src/getters/getConfigFromChain.ts]: What I did is I moved error handling "above" meaning that it's being handled wherever we are using this getter

File: src/network/getters/getTokenAndBalanceForSponsorship.ts (hunk 7 above)
Comment (mondoreale) [reply src/network/getters/getTokenAndBalanceForSponsorship.ts:22]: `getTokenInformation` uses a chain-specific public node. Chain the user is on should not matter.

**Pull request merged**